charts/schema-registry/templates/deployment.yaml

```diff
@@ -58,8 +58,29 @@ spec:
             mountPath: /etc/secret-volume
             readOnly: true
           {{- end }}
+          {{- if and .Values.container .Values.container.volume }}
+          - name: {{ .Values.container.volume.name }}
+            mountPath: {{ .Values.container.volume.mountPath }}
           {{- end }}
-      {{- if or .Values.authentication.enabled .Values.tls.enabled }}
+          {{- end }}
+      {{- if .Values.initContainer }}
+      initContainers:
+        - name: {{ .Values.initContainer.name }}
+          image: "{{ .Values.initContainer.image.repository }}:{{ .Values.initContainer.image.tag }}"
+          imagePullPolicy: {{ .Values.initContainer.image.pullPolicy }}
+          command:
+{{ toYaml .Values.initContainer.command | indent 12 }}
+          {{- if .Values.initContainer.resources }}
+          resources:
+{{ toYaml .Values.initContainer.command | indent 12 }}
+          {{- end }}
+          {{- if .Values.initContainer.volume }}
+          volumeMounts:
+          - name: {{ .Values.initContainer.volume.name }}
```

Comment:
Here, it is critical that the volume name that is used the same as the one defined at line 96, and the same used in the main container at line 62.    Else the volume is not shared, the and copy from initcontainer to main container will not work.

So basically I would have a shared-init-volume field in the values file outisde of the initcontainer and container sections.   The mount path however, should remain individual to the initcontainer and container like you have it.

Reply:
Removed the volume name from values.yaml